### PR TITLE
[stdlib][DNM] Experiment with a different array growth rate

### PR DESCRIPTION
**Pull request by airspeedswift**
Just a test to see if this makes much impact on the benchmarks.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by jckarter** [comment]
IIRC @dabrahams tried this early on, and 2x ended up winning. (Doesn't mean things haven't changed since then.)

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
That’ll teach me to try and code from the github editor.

File: stdlib/public/core/ArrayShared.swift [hunk 1]
```diff
@@ -143,7 +143,7 @@ internal func _expectEnd<C: Collection>(of s: C, is i: C.Index) {
 
 @inlinable
 internal func _growArrayCapacity(_ capacity: Int) -> Int {
-  return capacity * 2
+  return Int(Double(capacity) * 1.7)
```
Comment (jckarter) [inline stdlib/public/core/ArrayShared.swift]: `capacity * 17 / 10` is probably more efficient than round-tripping through the FPU. The constant multiply-divide can be replaced with a magic multiplication.

Comment (stephentyrone) [reply stdlib/public/core/ArrayShared.swift]: It doesn't matter much, except on cores without floating-point. Using 1 + 3/4 would be better, and basically equivalent. Also need to round-up (or special-case `capacity` == 1). Personally, I would probably use:
~~~~
return capacity + (capacity + 3)/4
~~~~

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/ArrayShared.swift (hunk 1 above)
Comment (airspeedswift) [reply stdlib/public/core/ArrayShared.swift]: > Also need to round-up (or special-case capacity == 1). 
🤦🏻‍♂️

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/ArrayShared.swift (hunk 1 above)
Comment (stephentyrone) [reply stdlib/public/core/ArrayShared.swift]: If it's any consolation, @natecook1000 managed to have exactly the same bug. Or as he put it, "implemented fixed-length arrays."

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Pull request closed without merging**